File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        string[] parts = logLine.Split(":");
        return parts[1].Trim();
    }

    public static string LogLevel(string logLine)
    {
        string[] parts = logLine.Split(":");
        return parts[0].Substring(1, parts[0].Length - 2).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine) + " (" + LogLevel(logLine) + ")";
    }
}

File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(':') + 1).strip();
    }

    public static String logLevel(String logLine) {
        String level = logLine.substring(1, logLine.indexOf(':') - 1);
        return level.toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
